### PR TITLE
Fixed {receipt_link} markup stripped in plain text emails - will send…

### LINKED ISSUE #4359: {receipt_link} markup stripped in plain text emails
When you select "No template, plain text only" and have `{receipt_link}` all you get is the non-clickable text. We should be checking for this within `edd_email_tag_receipt_link` and output URL if plain text so email clients can hyperlink it.

Reported via Give here: https://github.com/WordImpress/Give/issues/384

Fixed here: https://github.com/WordImpress/Give/pull/495

LMK if I can submit a PR if you agree.

### PR DESCRIPTION
**Pull request by DevinWalker**
Fixes #4359

### CODE REVIEW

**Comment by DevinWalker** [comment]
There we go.

File: includes/emails/class-edd-email-tags.php [hunk 1]
```diff
@@ -834,12 +834,22 @@ function edd_email_tag_sitename( $payment_id ) {
...
  *
  * @return string receipt_link
  */
 function edd_email_tag_receipt_link( $payment_id ) {
-	return sprintf( __( '%1$sView it in your browser.%2$s', 'easy-digital-downloads' ), '<a href="' . esc_url( add_query_arg( array( 'payment_key' => edd_get_payment_key( $payment_id ), 'edd_action' => 'view_receipt' ), home_url() ) ) . '">', '</a>' );
+	$receipt_url = esc_url( add_query_arg( array(
+		'payment_key' => edd_get_payment_key( $payment_id ),
+		'edd_action'  => 'view_receipt'
+	), home_url() ) );
+	$formatted   = sprintf( __( '%1$sView it in your browser %2$s', 'edd' ), '<a href="' . $receipt_url . '">', '&raquo;</a>' );
```
Comment (pippinsplugins) [inline includes/emails/class-edd-email-tags.php]: Can you set it up to run this only when the email template is not `none`? No need to run it otherwise.

**Comment by DevinWalker** [comment]
@pippinsplugins just updated per your request

**Comment by cklosowski** [comment]
Closing in favor of #4379 as it'll fix the unit tests...Your commit history was maintained @DevinWalker

**Comment by DevinWalker** [comment]
Cool thanks!

On Mon, Feb 29, 2016 at 5:59 PM, Chris Klosowski <notifications@github.com>
wrote:

> Closing in favor of #4379
> <https://github.com/easydigitaldownloads/Easy-Digital-Downloads/pull/4379>
> as it'll fix the unit tests...Your commit history was maintained
> @DevinWalker <https://github.com/DevinWalker>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/easydigitaldownloads/Easy-Digital-Downloads/pull/4371#issuecomment-190493332>
> .
>



-- 

Devin Walker

Founder & Lead Developer

P:

858-449-6460 <8584496460>

W:

http://wordimpress.com

433 G Street #201
San Diego, CA 92101 <https://goo.gl/maps/BePgp>
<https://www.facebook.com/wordimpressive>  <https://twitter.com/wordimpress>
  <https://plus.google.com/u/0/b/117062083910623146392/+Wordimpress/about>
<http://wordimpress.com/>

**Pull request closed without merging**